src/uid2Sdk.ts

```diff
@@ -156,6 +277,26 @@ export class UID2 {
     this._callbackManager?.runCallbacks(EventType.InitCompleted, {});
   }
 
+  private async loginInternal(opts: LoginOptions): Promise<void> {
+    if (!this._initComplete) {
+      throw new Error("Cannot login before calling init.");
+    }
+
+    const validatedNormalizedInput = UID2.validateAndNormalize(opts);
+    if (!validatedNormalizedInput) {
+      // TODO: Better error message.
+      throw new Error("Invalid DII");
```

Comment:
what about "Cannot login due to invalid input"?

Reply:
Or is this copy pasted from somewhere else as well 😅

Reply:
😅 no, this one I take responsibility for.